subsys/ipc/ipc_service/backends/ipc_rpmsg.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ipc/ipc_rpmsg.h>
+
+static void rpmsg_service_unbind(struct rpmsg_endpoint *ep)
+{
+	rpmsg_destroy_ept(ep);
+}
+
+static void ns_bind_cb(struct rpmsg_device *rdev, const char *name, uint32_t dest)
+{
+	struct rpmsg_virtio_device *p_rvdev;
+	struct ipc_rpmsg_instance *instance;
+	struct ipc_ept *ept;
+	int err;
+
+	p_rvdev = CONTAINER_OF(rdev, struct rpmsg_virtio_device, rdev);
+	instance = CONTAINER_OF(p_rvdev->shpool, struct ipc_rpmsg_instance, shm_pool);
+
+	for (size_t i = 0; i < NUM_ENDPOINTS; i++) {
+		ept = &instance->endpoint[i];
+
+		if (strcmp(name, ept->name) == 0) {
+			err = rpmsg_create_ept(&ept->ep, rdev, name, RPMSG_ADDR_ANY,
+					       dest, instance->cb, rpmsg_service_unbind);
+			if (err != 0) {
+				return;
+			}
+
+			ept->bound = true;
+			if (instance->bound_cb) {
+				instance->bound_cb(ept);
+			}
+		}
+	}
+}
+
+int ipc_rpmsg_register_ept(struct ipc_rpmsg_instance *instance, unsigned int role,
+			   struct ipc_ept *ept)
+{
+	struct rpmsg_device *rdev;
+	int err;
+
+	if (!instance || !ept) {
+		return -EINVAL;
+	}
+
+	rdev = rpmsg_virtio_get_rpmsg_device(&instance->rvdev);
+
+	if (role == RPMSG_REMOTE) {
+		err = rpmsg_create_ept(&ept->ep, rdev, ept->name, RPMSG_ADDR_ANY,
+				       RPMSG_ADDR_ANY, instance->cb, rpmsg_service_unbind);
+		if (err != 0) {
```

Comment:
Is this `if` necessary? It would be more readable to init `err = 0` at the beginning of the function and just do one `return err` at the end.